### ISSUE TITLE
Code python pour connaitre les poids et biais a partir de training.txt

### ISSUE THREAD

**Issue by angv5845** [angv5845/Projet_IA_calcul_embarque#8]
import re
import numpy as np
import tensorflow as tf
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Conv2D, MaxPooling2D, Flatten, Dense, InputLayer
from tensorflow.keras.optimizers import Adam

# === PARSE LE FICHIER .txt POUR EXTRAIRE LE TABLEAU C-STYLE ===
def parse_training_txt(file_path):
    with open(file_path, 'r') as f:
        txt = f.read()
    data_txt = re.findall(r'{([^{}]+)}', txt)
    values = [list(map(float, line.split(','))) for line in data_txt]
    array = np.array(values).reshape((100, 48, 13))
    return array

# === CHARGEMENT DES DONNÉES ===
X = parse_training_txt("training.txt")

# Les 50 premiers sont "Rouge" (classe 0), les 50 derniers "Bleu" (classe 1)
y = np.array([0]*50 + [1]*50)

# Ajouter une dimension channel=1
X = X[..., np.newaxis]

# === CONCEPTION DU MODELE CNN+MLP ===
model = Sequential([
    InputLayer(input_shape=(48, 13, 1)),
    Conv2D(4, kernel_size=(3, 3), activation='sigmoid'),
    MaxPooling2D(pool_size=(2, 2)),
    Flatten(),
    Dense(1, activation='sigmoid')
])

model.compile(optimizer=Adam(0.01), loss='mean_squared_error', metrics=['accuracy'])

# === ENTRAINEMENT ===
model.fit(X, y, epochs=20, batch_size=10, verbose=1)

# === EXTRACTION DES POIDS ET BIAIS POUR ARDUINO ===
def flatten_weights_biases(model):
    weights = []
    biases = []
    for layer in model.layers:
        if hasattr(layer, 'get_weights'):
            w = layer.get_weights()
            if len(w) == 2:
                weights.extend(w[0].flatten())
                biases.extend(w[1].flatten())
    return weights, biases

weights, biases = flatten_weights_biases(model)

# === EXPORT POUR ARDUINO ===
def format_cpp_array(name, arr):
    return f"float {name}[] = {{\n  " + ',\n  '.join(f"{v:.6f}" for v in arr) + "\n};"

with open("weights_arduino.txt", "w") as f:
    f.write(format_cpp_array("weights", weights) + "\n\n")
    f.write(format_cpp_array("biases", biases) + "\n")

print("✅ Poids et biais exportés dans weights_arduino.txt")
